FIO/local-volumes/dbench-job-generator-local-volume-replica.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+set -euo pipefail
+
+#
+# The following script provisions a local volume with one replica and runs fio tests
+# to measure StorageOS performance. The FIO tests that are run can be found
+# here: https://github.com/storageos/dbench/blob/master/docker-entrypoint.sh
+#
+# In order to successfully execute the tests you will need to have:
+#  - Kubernetes cluster with a minium of 3 nodes and 30 Gib space
+#  - kubectl in the PATH - kubectl access to this cluster with
+#    cluster-admin privileges - export KUBECONFIG as appropriate
+#  - StorageOS CLI running as a pod in the cluster
+#  - jq in the PATH
+#
+# Deploy the StorageOS CLI as a container:
+# $ kubectl -n kube-system run \
+# --image storageos/cli:v2.1.0 \
+# --restart=Never                          \
+# --env STORAGEOS_ENDPOINTS=storageos:5705 \
+# --env STORAGEOS_USERNAME=storageos       \
+# --env STORAGEOS_PASSWORD=storageos       \
+# --command cli                            \
+# -- /bin/sh -c "while true; do sleep 999999; done"
+#
+
+# Define some colours for later
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+echo -e "${GREEN}Scenario: Local Volume with a replica${NC}"
+echo
+
+# Checking if jq is in the PATH
+if ! command -v jq &> /dev/null
+then
+    echo -e "${RED}jq could not be found. Please install jq and run the script again${NC}"
+    exit 1
+fi
+
+# Get the node name and id where the volume will get provisioned and attached on
+# Using the StorageOS cli is guarantee that the node is running StorageOS
+node_details=$(kubectl -n kube-system exec cli -- storageos describe nodes -ojson | jq -r '[.[0].labels."kubernetes.io/hostname",.[0].id]')
+node_id=$(echo $node_details | jq -r '.[1]')
+node_name=$(echo $node_details | jq -r '.[0]')
+pvc_prefix="$RANDOM"
+manifest_path="./tmp-local-fio"
+fio_job="local-volume-with-replica-fio"
+manifest="${manifest_path}/${fio_job}.yaml"
+logs_path="./tmp-fio-logs"
+
+
```

Comment:
Can we shift all temporary file and directory creation to use mktemp(1) instead? It contains all of the appropriate checks and plumbing to create temp files and directories in a race free way. Then we can either clean up or drop a message to the console and say 'I've left temporary scripts and results in $TMPDIR - clean up when ready'.

If you use the 'template' function of mktemp then you can also include the pattern in .gitignore to preserve that behaviour.